nsxt/resource_nsxt_policy_shared_resource.go

```diff
@@ -0,0 +1,207 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/terraform-provider-nsxt/api/infra/shares"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func resourceNsxtPolicySharedResource() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicySharedResourceCreate,
+		Read:   resourceNsxtPolicySharedResourceRead,
+		Update: resourceNsxtPolicySharedResourceUpdate,
+		Delete: resourceNsxtPolicySharedResourceDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceNsxtPolicySharedResourceImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"context":      getContextSchema(false, false, false),
+			"share_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Share ID to associate the resource to",
+				ForceNew:    true,
+			},
+			"resource_object": {
+				Type:        schema.TypeList,
+				Description: "List of resources to be shared",
+				Required:    true,
+				MinItems:    1,
```

Comment:
I think we usually let NSX validate the number of items, but in this case I agree with doing it in the provider.
It is clearly pointless to create a shared resource without any resource :/